recipe/meta.yaml

```diff
@@ -44,11 +44,13 @@ requirements:
     - ninja
   host:
     ############################################################################################
-    # Pin to CCCL 2.7 to avoid build issues in older or newer CCCLs.                           #
+    # On PPC, pin to CCCL 2.7 for compatibility with CUDA 12.4. CCCL 2.8+ doesn't work there.  #
+    # Otherwise use the latest CCCL.                                                           #
     #                                                                                          #
     # xref: https://github.com/conda-forge/xgboost-feedstock/pull/219#issuecomment-2738813028  #
     ############################################################################################
-    - cccl =2.7             # [cuda_compiler_version != "None"]
+    - cccl =2.7             # [ppc64le and cuda_compiler_version != "None"]
+    - cccl                  # [not ppc64le and cuda_compiler_version != "None"]
```

Comment:
Looks like we still need this pin for `linux_ppc64le` as it uses CUDA 12.4, which isn't compatible with anything newer than CCCL 2.7. Tried both CCCL 2.8 and 3.0 in this PR, which both failed. An earlier version of CCCL 2.8 failed previously

Given all of this have readded the CCCL 2.7 pin for `linux_ppc64le`. Everything else builds ok on CCCL 3. So have let the rest float. Currently we are picking up CCCL 3.0.2